opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/HwEntityDaoHibernate.java

```diff
@@ -55,6 +64,72 @@ public OnmsHwEntity findRootByNodeId(Integer nodeId) {
         return (OnmsHwEntity) findUnique("from OnmsHwEntity e where e.parent is null and e.node.id = ?", nodeId);
     }
 
+    @Override
+    public HwEntity findRootEntityByNodeId(Integer nodeId) {
+        List<HwEntity> entityList = getHibernateTemplate().execute(session -> (List<HwEntity>)
+                session.createSQLQuery("select * FROM hwEntity where nodeid = " + nodeId)
+                        .setResultTransformer(
+                                new ResultTransformer() {
+                                    @Override
+                                    public Object transformTuple(Object[] tuple, String[] strings) {
+                                        return new HwEntity((Integer) tuple[0], (Integer) tuple[1], (Integer) tuple[2], (Integer) tuple[3], (Integer) tuple[4],
+                                                (String) tuple[5], (String) tuple[6], (String) tuple[7], (String) tuple[8], (String) tuple[9],
+                                                (String) tuple[10], (String) tuple[11], (String) tuple[12], (String) tuple[13], (String) tuple[14],
+                                                (String) tuple[15], (String) tuple[16], (Boolean) tuple[17], (Date) tuple[18], (String) tuple[19]);
+                                    }
+
+                                    @Override
+                                    public List transformList(List list) {
+                                        return list;
+                                    }
+                                }
+                        ).list());
+        Optional<HwEntity> optionalHwEntity =  entityList.stream().filter(hwEntity -> hwEntity.getParentId() == null).findFirst();
+        if(optionalHwEntity.isPresent()) {
+            HwEntity parent = optionalHwEntity.get();
+            List<HwEntityAlias> hwEntityAliases = findHwEntityAlias(parent);
+            parent.addHwEntityAliasList(hwEntityAliases);
+            findChildren(parent, entityList);
+            return parent;
+        }
+        return null;
+    }
+
+    private void findChildren(HwEntity parent, List<HwEntity> hwEntityList) {
+
+        Set<HwEntity> children = hwEntityList.stream().filter(hwEntity -> {
+                    if (hwEntity.getParentId() != null) {
+                        return hwEntity.getParentId().equals(parent.getId());
+                    }
+                    return false;
+                }
+        ).collect(Collectors.toSet());
```

Comment:
This section can be simplified as the following 
`Set<HwEntity> children = hwEntityList.stream().filter(hwEntity -> hwEntity.getParentId()!=null && hwEntity.getParentId().equals(parent.getId())).collect(Collectors.toSet());`